### PR TITLE
T337002 | Hooks cleanup

### PR DESCRIPTION
**Pull request by maaartyyynaa**
Hooks cleanup:

- switch to HookHandler
- inject services via constructor

### CODE REVIEW

File: extension.json [hunk 1]
```diff
@@ -177,19 +177,29 @@
...
+		"BeforePageDisplay": "main",
+		"DiffTools": "main",
+		"DifferenceEngineViewHeader": "main",
+		"GetAllBlockActions": "main",
+		"GetLogTypesOnUser": "main",
+		"HistoryTools": "main",
+		"LocalUserCreated": "main",
+		"LogEventsListLineEnding": "main",
+		"PageHistoryBeforeList": "main",
 		"BeforeCreateEchoEvent": "MediaWiki\\Extension\\Thanks\\Hooks::onBeforeCreateEchoEvent",
```
Comment (emkarcinos) [inline extension.json:190]: You can use normal methods instead of static ones even though there's no interface

File: extension.json [hunk 2]
```diff
@@ -177,19 +177,29 @@
...
+		"EchoGetBundleRules": "MediaWiki\\Extension\\Thanks\\Hooks::onEchoGetBundleRules"
+	},
+	"HookHandlers":{
+		"main":{
+			"class":"MediaWiki\\Extension\\Thanks\\Hooks",
+			"services": [
+				"RevisionLookup",
+				"UserOptionsManager",
+				"MainConfig"
+			]
```
Comment (emkarcinos) [inline extension.json:201]: **suggestion** styling
```suggestion
	"HookHandlers": {
		"main": {
			"class": "MediaWiki\\Extension\\Thanks\\Hooks",
			"services": [
				"RevisionLookup",
				"UserOptionsManager",
				"MainConfig"
			]
```

File: includes/Hooks.php [hunk 3]
```diff
@@ -35,187 +45,79 @@
...
+    DifferenceEngineViewHeaderHook,
+    GetAllBlockActionsHook,
+    GetLogTypesOnUserHook,
+    HistoryToolsHook,
+    LocalUserCreatedHook,
+    LogEventsListLineEndingHook,
+    PageHistoryBeforeListHook
+{
+
+    /** @var RevisionLookup $revisionLookup */
```
Comment (emkarcinos) [inline includes/Hooks.php:61]: I think those commits can be omitted, types are specified anyways so we don't really need them

File: includes/Hooks.php [hunk 4]
```diff
@@ -453,27 +354,26 @@ public static function onEchoGetBundleRules( $event, &$bundleString ) {
 	 * @param string[] &$attribs HTML attributes to add to the line.
 	 * @throws ConfigException
 	 */
-	public static function onLogEventsListLineEnding(
-		LogEventsList $page, &$ret, DatabaseLogEntry $entry, &$classes, &$attribs
-	) {
+    public function onLogEventsListLineEnding( $page, &$ret, $entry, &$classes,
+                                               &$attribs
+    ) {
```
Comment (emkarcinos) [inline includes/Hooks.php:359]: **suggestion**: styling
```suggestion
    public function onLogEventsListLineEnding(
        $page,
        &$ret,
        $entry,
        &$classes,
        &$attribs
    ) {
```

File: includes/Hooks.php [hunk 5]
```diff
@@ -453,27 +354,26 @@ public static function onEchoGetBundleRules( $event, &$bundleString ) {
...
+			|| HooksHelper::isUserBlockedFromThanks( $user )
 		) {
 			return;
 		}
 
 		// Make sure this log type is allowed.
-		$allowedLogTypes = MediaWikiServices::getInstance()
-			->getMainConfig()
+		$allowedLogTypes = $this->mainConfig
 			->get( 'ThanksAllowedLogTypes' );
```
Comment (emkarcinos) [inline includes/Hooks.php:374]: Use ServiceOptions instead

File: includes/Hooks.php [hunk 6]
```diff
@@ -453,27 +354,26 @@ public static function onEchoGetBundleRules( $event, &$bundleString ) {
...
 		if ( !in_array( $entry->getType(), $allowedLogTypes )
-			&& !in_array( $entry->getType() . '/' . $entry->getSubtype(), $allowedLogTypes ) ) {
+			&& !in_array( $entry->getType() . 'Hooks.php/' . $entry->getSubtype(), $allowedLogTypes ) ) {
```
Comment (emkarcinos) [inline includes/Hooks.php:376]: Typo?
```suggestion
			&& !in_array( $entry->getType() . '/' . $entry->getSubtype(), $allowedLogTypes ) ) {
```

**Comment by maaartyyynaa** [comment]
closing @emkarcinos cherry-picked those changes into his branch

**Pull request closed without merging**